src/common/Browser.ts

```diff
@@ -433,7 +433,7 @@ export class Browser extends EventEmitter {
       url: 'about:blank',
       browserContextId: contextId || undefined,
     });
-    const target = this._targets.get(targetId);
+    const target = await this._targets.get(targetId);
```

Comment:
Not sure whether you've looked into the source code but `this._targets` is a map of `Target` instances. The assignment happens here: 

https://github.com/puppeteer/puppeteer/blob/44b22bbc2629e3c75c1494b299a66790b371fb0a/src/common/Browser.ts#L350-L361

The target's constructor might be doing some async work, but you can't mark a constructor as `async` and await the class instantiation (i. e. `await new Target(...)` is not possible). The instance will be immediately returned and the constructor's async work happens afterwards in the background.

Did you actually try building puppeteer with this change to see whether this fixes the issue for you?